### PR TITLE
Issue #431: Remove observation process function

### LINKED ISSUE #431: Remove `observe_process()`
In PR #390 we updated the preprocessing functionality, but we left `observe_process()` in the package. I think if possible we should get rid of this. Similarly, if possible we should get rid of the `filter_obs` functions.

### PR DESCRIPTION
**Pull request by athowes**
## Description

This PR closes #431.

It moves the mutating calls into the tests, vignettes, and sticker. It also slims them down to what is needed.

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by athowes** [comment]
image

Failing here but it already has `tolerance = 0.2` so should be within range..

**Comment by seabbs** [comment]
I think tolerance is a relative measure more than you think and so needs maybe some tuning

**Review by seabbs — APPROVED** [review]
LGTM

**Review by seabbs — APPROVED** [review]

File: inst/make_hexsticker.R [hunk 1]
```diff
@@ -15,11 +15,19 @@ obs <- outbreak |>
...
+    stime_lwr = floor(.data$stime),
+    stime_upr = .data$stime_lwr + 1,
+    delay_daily = .data$stime_lwr - .data$ptime_lwr
+  )
 
 obs_time <- 25
+
 truncated_obs <- obs |>
-  filter(.data$stime_upr <= obs_time) |>
+  mutate(obs_time = obs_time) |>
```
Comment (seabbs) [inline inst/make_hexsticker.R:29]: why add this line?

File: tests/testthat/setup.R [hunk 2]
```diff
@@ -14,8 +14,14 @@ sim_obs <- simulate_gillespie() |>
...
     sdlog = sdlog
   ) |>
-  observe_process() |>
-  dplyr::filter(.data$stime_upr <= obs_time) |>
+  dplyr::mutate(
+    ptime_lwr = floor(.data$ptime),
+    ptime_upr = .data$ptime_lwr + 1,
+    stime_lwr = floor(.data$stime),
+    stime_upr = .data$stime_lwr + 1,
+    obs_time = obs_time
```
Comment (seabbs) [inline tests/testthat/setup.R:22]: why add obs_time vs using as a variable as previously?

File: inst/make_hexsticker.R (hunk 1 above)
Comment (athowes) [reply inst/make_hexsticker.R:29]: It's required I think

File: tests/testthat/setup.R (hunk 2 above)
Comment (athowes) [reply tests/testthat/setup.R:22]: I think later on in the code it is used as a part of the dataframe (this is what `observe_process` was doing, so previously it was not used as a variable I think)

**Comment by athowes** [comment]
If we get this merged I can work on #430.

File: tests/testthat/setup.R (hunk 2 above)
Comment (seabbs) [reply tests/testthat/setup.R:22]: its coded as a variable on  line 3 - if you just take it out of the mutate call I think it should all just work

**Review by seabbs — APPROVED** [review]

File: tests/testthat/setup.R (hunk 2 above)
Comment (athowes) [reply tests/testthat/setup.R:22]: In tests perhaps `obs_time` is needed in the `data.frame` (not only for the filtering)

**Pull request merged**